File: 1-arrays/easy/1-contains-duplicate-lc-217/java/contains-duplicate.java
import java.util.Set;
import java.util.HashSet;

class Solution {
    public boolean containsDuplicate([]int nums) {
        Set<Integer> set = new HashSet<>();

        for (i = 0; i < nums.length; i++) {
            if (set.contains(nums[i])) {
                return true;
            } else {
                set.add(nums[i]);
            }
        }
        return false;
    }
}

File: 1-arrays/easy/1-contains-duplicate-lc-217/cpp/contains-duplicate.cpp
#include <vector>
#include <unordered_set>
using std::unordered_set;
using std::vector;

class Solution {
    public bool containsDuplicate(vector<int> &nums) {
        unordered_set<int> set;

        for (i = 0; i < nums.size(); i++) {
            if (set.find(nums[i]) != set.end()) {
                return true;
            } else {
                set.insert(nums[i]);
            }
        }
        return false;
    }
}